pkg/util/ip_reservation_test.go

```diff
@@ -126,8 +174,42 @@ func TestGetUnReservedIPRange(t *testing.T) {
 			errorExpected: false,
 		},
 		{
-			name: "1 Pending 1 Used",
-			cidr: "192.168.92.0/27",
+			name:            "1 /29 Pending 0 Used enterprise",
+			cidr:            "192.168.92.0/22",
+			ipRangeSize:     IpRangeSizeEnterprise,
+			pendingIPRanges: make(map[string]bool),
+			cloudProviderReservedIPRanges: map[string]bool{
+				ips[3]: true,
+			},
+			expected:      "192.168.93.0/24",
+			errorExpected: false,
+		},
+		{
+			name:            "1 /24 Pending 0 Used",
+			cidr:            "192.168.92.0/22",
+			ipRangeSize:     IpRangeSize,
+			pendingIPRanges: make(map[string]bool),
+			cloudProviderReservedIPRanges: map[string]bool{
+				ipsLarge[0]: true,
+			},
+			expected:      "192.168.93.0/29",
+			errorExpected: false,
+		},
+		{
+			name:            "1 /24 Pending 0 Used enterprise",
```

Comment:
test case name says 1 /24 pending, but `cloudProviderReservedIPRanges` has the range range added. Is this correct? same with the other non zero pending test cases.